packages/core/src/client/index.ts

```diff
@@ -0,0 +1,413 @@
+    account: undefined,
+  });
+
+  public constructor(
+    private core: ICore,
+    private notifyClient: NotifyClient,
+    private domain: string
+  ) {}
+
+  //TODO: Make more efficient - this is very slow.
+  private static updateMessages() {
+    Web3InboxClient.subscriptionState.messages =
+      Web3InboxClient.instance?.notifyClient.messages
+        .getAll()
+        .filter((m) => m)
+        .map((m) => Object.values(m?.messages))
+        .flat()
+        .flat() ?? [];
+  }
+
+  protected attachEventListeners(): void {
+    const updateInternalSubscriptions = () => {
+      Web3InboxClient.subscriptionState.subscriptions =
+        this.notifyClient.subscriptions.getAll();
+    };
+
+    this.notifyClient.on("notify_message", Web3InboxClient.updateMessages);
+    this.notifyClient.on("notify_delete", updateInternalSubscriptions);
+    this.notifyClient.on(
+      "notify_subscriptions_changed",
+      updateInternalSubscriptions
+    );
+  }
+
+  public static getInstance(): Web3InboxClient {
+    if (Web3InboxClient.clientState.isReady && Web3InboxClient.instance) {
+      return Web3InboxClient.instance;
+    } else {
+      throw new Error(
+        "Client not ready. getInstance can only be called when client is ready. Consider using getIsReady"
+      );
+    }
+  }
+
+  public static getIsReady(): boolean {
+    return Web3InboxClient.clientState.isReady;
+  }
+
+  public static watchIsReady(cb: (isReady: boolean) => void) {
+    return subscribe(Web3InboxClient.clientState, () => {
+      cb(Web3InboxClient.clientState.isReady);
+    });
+  }
+
+  public setAccount(account: string) {
+    Web3InboxClient.clientState.account = account;
+  }
+
+  public getAccount() {
+    return Web3InboxClient.clientState.account;
+  }
+
+  public watchAccount(cb: (acc: string) => void) {
+    const acc = Web3InboxClient.clientState.account;
+    if (!acc) return;
+    return subscribe(Web3InboxClient.clientState, () => {
+      return cb(acc);
+    });
+  }
+
+  // initializes the client with persisted storage and a network connection
+  public static async init(params: {
+    projectId: string;
+    domain?: string;
+  }): Promise<Web3InboxClient> {
+    if (Web3InboxClient.clientState.initting) {
+      return new Promise<Web3InboxClient>((res) => {
+        subscribe(Web3InboxClient.clientState, () => {
+          if (!Web3InboxClient.clientState.isReady) {
+            res(Web3InboxClient.instance!);
+          }
+        });
+      });
+    }
+
+    if (Web3InboxClient.instance) return Web3InboxClient.instance;
+
+    Web3InboxClient.clientState.initting = true;
+
+    const core = new Core(params);
+
+    const identityKeys = new IdentityKeys(core, DEFAULT_KEYSERVER_URL);
+    const notifyParams = {
+      core,
+      identityKeys,
+      keyserverUrl: DEFAULT_KEYSERVER_URL,
+      projectId: params.projectId,
+      logger: "debug",
+    };
+
+    const notifyClient = await NotifyClient.init({
+      ...notifyParams,
+    });
```

Comment:
Nit: we can avoid a copy here
```suggestion
    const notifyClient = await NotifyClient.init(notifyParams);
```